integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,263 @@
+
+			for _, gotContainer := range containers {
+				normalize(gotContainer)
+				if reflect.DeepEqual(gotContainer, expectedContainer) {
+					return nil
+				}
+			}
+
+			return errors.New("couldn't find matching container")
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		BusyboxPodRepeatCommand(ns, "sleep inf"),
+		WaitUntilTestPodReadyCommand(ns),
+		listContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
+
+	runCommands(commands, t)
+}
+
+func TestListSingleContainer(t *testing.T) {
+	// TODO: Handle it once we support getting K8s container name for docker
+	if *containerRuntime == ContainerRuntimeDocker {
+		t.Skip("Skip TestListSingleContainer on docker since we don't propogate the Kubernetes pod container name")
+	}
+
+	podName := "test-pod"
+	containerName := "test-pod"
+	ns := GenerateTestNamespaceName("test-list-single-container")
+
+	t.Parallel()
+
+	listContainersCmd := &Command{
+		Name:         "RunListSingleContainer",
+		Cmd:          fmt.Sprintf("local-gadget list-containers -o json --runtimes=%s --containername=%s", *containerRuntime, containerName),
+		StartAndStop: true,
+		ExpectedOutputFn: func(output string) error {
+			expectedContainer := &containercollection.Container{
+				Name:      containerName,
+				Podname:   podName,
+				Runtime:   *containerRuntime,
+				Namespace: ns,
+			}
+
+			normalize := func(c *containercollection.Container) {
+				c.ID = ""
+				c.Pid = 0
+				c.OciConfig = nil
+				c.Bundle = ""
+				c.Mntns = 0
+				c.Netns = 0
+				c.CgroupPath = ""
+				c.CgroupID = 0
+				c.CgroupV1 = ""
+				c.CgroupV2 = ""
+				c.Labels = nil
+				c.PodUID = ""
+			}
+
+			var containers []*containercollection.Container
+			if err := json.Unmarshal([]byte(output), &containers); err != nil {
+				return err
+			}
+
+			for _, gotContainer := range containers {
+				normalize(gotContainer)
+				if !reflect.DeepEqual(gotContainer, expectedContainer) {
+					return fmt.Errorf("expect all containers to match, expected=%+v, got=%+v", gotContainer, expectedContainer)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		BusyboxPodRepeatCommand(ns, "sleep inf"),
+		WaitUntilTestPodReadyCommand(ns),
+		listContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
+
+	runCommands(commands, t)
+}
+
+func TestTraceTCPBind(t *testing.T) {
+	ns := GenerateTestNamespaceName("test-trace-tcpbind")
+
+	t.Parallel()
+
+	traceTCPBindCmd := &Command{
+		Name:         "TraceTCPBind",
+		Cmd:          fmt.Sprintf("local-gadget trace bind -o json --runtimes=%s", *containerRuntime),
+		StartAndStop: true,
+		ExpectedOutputFn: func(output string) error {
+			expectedEntry := &bindTypes.Event{
+				Event:    BuildBaseEvent(ns),
+				Comm:     "nc",
+				Protocol: "TCP",
+				Addr:     "::",
+				Port:     9090,
+				Options:  ".R...",
+			}
+
+			normalize := func(e *bindTypes.Event) {
+				// TODO: Handle it once we support getting K8s container name for docker
```

Comment:
Once we have the Kubernetes metadata available, I think we should start considering moving the shared code between inspektor-gadget and local-gadget to a common place. For instance, this test is identical except for the `Cmd` that calls `kubectl-gadget` or `local-gadget`.

Also because I expect to add more tests like this for each gadget that will be identical again. For example, to verify that the flag `--pid`, `--ports` and `--ignore-errors` are working properly.

cc @mauriciovasquezbernal

Reply:
Yes, I agree. In this case we're testing Kubernetes containers I expected both tests to be almost identical.